netcheck/netcheck.go

```diff
@@ -454,11 +458,63 @@ func (c *Client) GetReport(ctx context.Context) (*Report, error) {
 		}
 	}
 
-	// TODO: if UDP is blocked, try to measure TCP connect times
-	// to DERP nodes instead? So UDP-blocked users still get a
-	// decent DERP node, rather than being randomly assigned to
-	// the other side of the planet? Or try ICMP? (likely also
-	// blocked?)
+	// Try HTTPS if UDP latency check failed for every stun server
+	// TODO: It does not show that UDP latency check failed
+	need := stuns4
+	if pc6 != nil {
+		need = append(need, stuns6...)
+	}
+	var wg sync.WaitGroup // For HTTPS checks
+
+	for _, server := range need {
+		if _, ok := ret.DERPLatency[server]; !ok {
+			server := server
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+
+				c.logf("netcheck: %s UDP latency check failed, try HTTPS", server)
+				sp := strings.Split(server, ":")
```

Comment:
this doesn't work for IPv6 literals like `[2001:fe82::1]:80`. Use https://golang.org/pkg/net/#SplitHostPort instead.